samcli/lib/remote_invoke/lambda_invoke_executors.py

```diff
@@ -0,0 +1,143 @@
+                )
+                raise InvalidResourceBotoParameterException(error_msg)
+            else:
+                self.request_parameters[parameter_key] = parameter_value
+
+    def _execute_action(self, payload: str):
+        self.request_parameters["FunctionName"] = self._function_name
+        self.request_parameters["Payload"] = payload
+        LOG.debug(
+            "Calling lambda_client.invoke with FunctionName:%s, Payload:%s, parameters:%s",
+            self._function_name,
+            payload,
+            self.request_parameters,
+        )
+        return self._lambda_client.invoke(**self.request_parameters)
+
+
+class DefaultConvertToJSON(RemoteInvokeRequestResponseMapper):
+    """
+    If a regular string is provided as payload, this class will convert it into a JSON object
+    """
+
+    def map(self, test_input: RemoteInvokeExecutionInfo) -> RemoteInvokeExecutionInfo:
+        if not test_input.is_file_provided():
+            try:
+                _ = json.loads(cast(str, test_input.payload))
+            except JSONDecodeError:
+                json_value = json.dumps(test_input.payload)
+                LOG.info(
+                    "Auto converting value '%s' into JSON '%s'. "
+                    "If you don't want auto-conversion, please provide a JSON string as payload",
+                    test_input.payload,
+                    json_value,
+                )
+                test_input.payload = json_value
+
+        return test_input
+
+
+class LambdaResponseConverter(RemoteInvokeRequestResponseMapper):
+    """
+    This class helps to convert response from lambda service. Normally lambda service
+    returns 'Payload' field as stream, this class converts that stream into string object
+    """
+
+    def map(self, remote_invoke_input: RemoteInvokeExecutionInfo) -> RemoteInvokeExecutionInfo:
+        assert isinstance(remote_invoke_input.response, dict), "Invalid response type received, expecting dict"
+        payload_field = remote_invoke_input.response.get("Payload")
+        if payload_field:
+            remote_invoke_input.response["Payload"] = cast(StreamingBody, payload_field).read().decode("utf-8")
+
+        return remote_invoke_input
+
+
+class LambdaResponseOutputFormatter(RemoteInvokeRequestResponseMapper):
+    """
+    This class helps to format output response for lambda service that will be printed on the CLI.
+    If LogResult is found in the response, the decoded LogResult will be written to stderr. The response payload will
+    be written to stdout.
+    """
+
+    def map(self, remote_invoke_input: RemoteInvokeExecutionInfo) -> RemoteInvokeExecutionInfo:
+        """
+        Maps the lambda response output to the type of output format specified by customer.
+        If output_format is original-boto-response, write the original boto repsonse
+        to stdout.
+        """
+        if remote_invoke_input.output_format == "default":
```

Comment:
Should these output formats be an enum so that we don't need to copy/paste string constants in different places?